subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy

```diff
@@ -20,69 +20,92 @@ class KnownProblemIds {
-        'dependency-version-catalog:alias-not-finished',
-        'dependency-version-catalog:invalid-dependency-notation',
-        'dependency-version-catalog:reserved-alias-name',
-        'dependency-version-catalog:catalog-file-does-not-exist',
-        'dependency-version-catalog:toml-syntax-error',
-        'dependency-version-catalog:too-many-import-files',
-        'dependency-version-catalog:too-many-import-invocation',
-        'dependency-version-catalog:no-import-files',
-        'deprecation:deprecated-feature-used',
-        'task-selection:no-matches',
-        'validation:property-validation:annotation-invalid-in-context',
-        'validation:property-validation:cannot-use-optional-on-primitive-types',
-        'validation:property-validation:cannot-write-output',
-        'validation:property-validation:conflicting-annotations',
-        'validation:property-validation:ignored-property-must-not-be-annotated',
-        'validation:property-validation:implicit-dependency',
-        'validation:property-validation:incompatible-annotations',
-        'validation:property-validation:incorrect-use-of-input-annotation',
-        'validation:property-validation:input-file-does-not-exist',
-        'validation:property-validation:missing-annotation',
-        'validation:property-validation:missing-normalization-annotation',
-        'validation:property-validation:nested-map-unsupported-key-type',
-        'validation:property-validation:nested-type-unsupported',
-        'validation:property-validation:mutable-type-with-setter',
-        'validation:property-validation:private-getter-must-not-be-annotated',
-        'validation:property-validation:unexpected-input-file-type',
-        'validation:property-validation:unsupported-notation',
-        'validation:property-validation:unknown-implementation',
-        'validation:property-validation:unsupported-value-type',
-        'validation:property-validation:value-not-set',
-        'validation:type-validation:ignored-annotations-on-method',
-        'validation:type-validation:invalid-use-of-type-annotation',
-        'validation:type-validation:not-cacheable-without-reason',
+    private static final def KNOWN_GROUPS = [
+        'problems-api' : 'Problems API',
+        'validation' : 'Validation',
+        'compilation' : 'Compilation',
+        'deprecation' : 'Deprecation',
+        'compilation:java' : 'Java compilation',
+        'task-selection' : 'Task selection',
+        'dependency-version-catalog' : 'Version catalog',
+        'compilation:groovy-dsl' : 'Groovy DSL script compilation',
+        'validation:property-validation' : 'Property validation problems',
+        'validation:type-validation' : 'Gradle type validation',
+
+        // groups from integration tests
+        'generic' : 'Generic'
+    ]
+
+    private static final def KNOWN_DEFINITIONS = [
+        'problems-api:missing-id' : 'Problem id must be specified',
```

Comment:
Do we still need the "known definitions" when we check all problems anyway with verifyAll?

Reply:
I think so. The known definitions fixture should be used to automatically check the problem definition. Anything we do in `verifyAll` is to validate the context. 

Fyi, `verifyAll` is a Spock concept that will verify all assertions in the closure at once, instead of failing on the first false statement.